# Exploratory Data Analysis on Salaries

In [2]:
# Importing numpy and pandas library 
import numpy as np
import pandas as pd

# Reading required data as a dataframe, and assigning it a variable name
df = pd.read_csv('Salaries.csv')

In [3]:
# Inspecting the first 5 entries of the dataframe
df.head()

rank discipline  phd  service   sex  salary
0  Prof          B   56       49  Male  186960
1  Prof          A   12        6  Male   93000
2  Prof          A   23       20  Male  110515
3  Prof          A   40       31  Male  131205
4  Prof          B   20       18  Male  104800

In [4]:
# Inspecting the data types in each column of the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   rank        78 non-null     object
 1   discipline  78 non-null     object
 2   phd         78 non-null     int64 
 3   service     78 non-null     int64 
 4   sex         78 non-null     object
 5   salary      78 non-null     int64 
dtypes: int64(3), object(3)
memory usage: 3.8+ KB


In [5]:
# Creating a summary statistics of the dataframe, rounded to 2
round(df.describe(), 2)

phd  service     salary
count  78.00    78.00      78.00
mean   19.71    15.05  108023.78
std    12.50    12.14   28293.66
min     1.00     0.00   57800.00
25%    10.25     5.25   88612.50
50%    18.50    14.50  104671.00
75%    27.75    20.75  126774.75
max    56.00    51.00  186960.00

Deductions from above output:
1. there are 78 entries in the dataset
2. the average phd is 19.71, service is 15.05, and salary is 108023.78
3. the minimum phd is 1, service is 0, and salary is 57800.00
4. the maximum phd is 56, service is 51, and salary is 186960.00
5. less than 25% of individuals in our dataset have 10.25 phd, 5.25 service, and earns 88612.50
6. less than 50% of individuals in our dataset have 18.50 phd, 14.50 service, and earns 104671.00
7. less than 75% of individuals in our dataset have 27.75 phd, 20.75 service, and earns 126774.75

In [6]:
# Creating summary statistics of the salary column, grouped by rank, and rounded to 2 
round(df.groupby(['rank'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]),2)

salary                          
           count       mean    min     max
rank                                      
Prof          46  123624.80  57800  186960
AssocProf     13   91786.23  62884  119800
AsstProf      19   81362.79  63100   97032

Deductions from above output:
Individuals with the rank of "Prof" who are 46 in count, earn more than their colleagues on the average and maximum salary scale, but also some of them who earn lesser on the minimum salary scale.

In [7]:
# Creating summary statistics of the salary column, grouped by discipine, and rounded to 2 
round(df.groupby(['discipline'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]),2)

salary                          
            count       mean    min     max
discipline                                 
B              42  116331.79  71065  186960
A              36   98331.11  57800  155865

Deductions from above output:
Individuals with discipline B who are 42 in count, earn more than their counterparts in discipline A in all metrics of the salary scale.

In [8]:
# Creating summary statistics of the salary column, grouped by sex, and rounded to 2 
round(df.groupby(['sex'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]),2)

salary                          
        count       mean    min     max
sex                                    
Male       39  115045.15  57800  186960
Female     39  101002.41  62884  161101

Deductions from above output:
Male individuals earn more than their female colleagues on the average and maximum salary scale, but earn lower than them on the minimum. 

In [9]:
# Creating summary statistics of individuals who earn salaries less than or equal to 100,000, grouped by rank, and rounded to 2 
round(df[df['salary'] >= 100000].groupby(['rank'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]), 2)

salary                           
           count       mean     min     max
rank                                       
Prof          39  129990.33  102000  186960
AssocProf      7  107479.57  103613  119800

Deductions from above output:
1. 39 individuals of the Prof rank earn more than 100,000 with an average of 129,990.33, minimum of 102,000, and maximum of 186,960.
2. 7 individuals of the AssocProf rank earn more than 100,000 with an average of 107,479.57, minimum of 103,613, and maximum of 119,800.

In [10]:
# Creating summary statistics of individuals who earn salaries less than or equal to 100,000, grouped by discipline, and rounded to 2 
round(df[df['salary'] >= 100000].groupby(['discipline'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]), 2)

salary                           
            count       mean     min     max
discipline                                  
B              29  129833.07  103613  186960
A              17  120989.47  102000  155865

Deductions from above output:
1. 29 individuals of the discipline B earn more than 100,000 with an average of 129,833.07, minimum of 103,613, and maximum of 186,960.
2. 17 individuals of the discipline A earn more than 100,000 with an average of 120,989.47, minimum of 102,000, and maximum of 155,865.

In [11]:
# Creating summary statistics of individuals who earn salaries less than or equal to 100,000, grouped by sex, and rounded to 2 
round(df[df['salary'] >= 100000].groupby(['sex'], sort=False)[['salary']].agg(["count", "mean", "min", "max",]), 2)

salary                           
        count       mean     min     max
sex                                     
Male       25  131298.56  102000  186960
Female     21  120929.33  103613  161101

Deductions from above output:
1. 25 male individuals earn more than 100,000 with an average of 131,298.56, minimum of 102,000, and maximum of 186,960.
2. 21 female individuals earn more than 100,000 with an average of 120,929.33, minimum of 103,613, and maximum of 161,101.

In [12]:
# Creating summary statistics of individuals with phd, grouped by rank, and rounded to 2 
round(df.groupby(['rank'], sort=False)[['phd']].agg(["count", "mean", "min", "max",]),2)

phd               
          count   mean min max
rank                          
Prof         46  27.07  12  56
AssocProf    13  15.08   9  26
AsstProf     19   5.05   1  11

Deductions from above output:
1. 46 individuals of the Prof rank have on the average, minimum, and maximum 27.07, 12, and 56 phds respectively
2. 13 individuals of the AssocProf rank have on the average, minimum, and maximum 15.08, 9, and 26 phds respectively
3. 19 individuals of the AsstProf rank have on the average, minimum, and maximum 5.057, 1, and 11 phds respectively

In [13]:
# Creating summary statistics of individuals with phd, grouped by discipline, and rounded to 2 
round(df.groupby(['discipline'], sort=False)[['phd']].agg(["count", "mean", "min", "max",]),2)

phd               
           count   mean min max
discipline                     
B             42  18.14   1  56
A             36  21.53   2  51

Deductions from above output:
1. 42 individuals of the discipline B have on the average, minimum, and maximum 18.14, 1, and 56 phds respectively
2. 36 individuals of the discipline A have on the average, minimum, and maximum 21.53, 2, and 51 phds respectively

In [14]:
# Creating summary statistics of individuals with phd, grouped by sex, and rounded to 2 
round(df.groupby(['sex'], sort=False)[['phd']].agg(["count", "mean", "min", "max",]),2)

phd               
       count   mean min max
sex                        
Male      39  22.90   1  56
Female    39  16.51   2  39

Deductions from above output:
1. 39 male individuals have on the average, minimum, and maximum 22.90, 1, and 56 phds respectively
2. 39 female individuals have on the average, minimum, and maximum 16.51, 2, and 39 phds respectively

In [15]:
# Creating summary statistics of individuals with years in service, grouped by rank, and rounded to 2 
round(df.groupby(['rank'], sort=False)[['service']].agg(["count", "mean", "min", "max",]),2)

service               
            count   mean min max
rank                            
Prof           46  21.41   0  51
AssocProf      13  11.31   6  24
AsstProf       19   2.21   0   6

Deductions from above output:
1. 46 individuals of the Prof rank have on the average, minimum, and maximum 21.41, 0, and 51 years in service respectively
2. 13 individuals of the AssocProf rank have on the average, minimum, and maximum 11.31, 6, 24 years in service respectively
3. 19 individuals of the AsstProf rank have on the average, minimum, and maximum 2.21, 0, 6 years in service respectively

In [16]:
# Creating summary statistics of individuals with years in service, grouped by discipline, and rounded to 2 
round(df.groupby(['discipline'], sort=False)[['service']].agg(["count", "mean", "min", "max",]),2)

service               
             count   mean min max
discipline                       
B               42  14.48   0  49
A               36  15.72   0  51

Deductions from above output:
1. 42 individuals of the discipline A have on the average, minimum, and maximum 14.48, 0, and 49 years in service respectively
2. 36 individuals of the discipline B have on the average, minimum, and maximum 15.72, 0, 51 years in service respectively


In [17]:
# Creating summary statistics of individuals with years in service, grouped by sex, and rounded to 2 
round(df.groupby(['sex'], sort=False)[['service']].agg(["count", "mean", "min", "max",]),2)

service               
         count   mean min max
sex                          
Male        39  18.54   0  51
Female      39  11.56   0  36

Deductions from above output:
1. 39 male individuals have on the average, minimum, and maximum 18.54, 0, and 51 years in service respectively
2. 39 female individuals have on the average, minimum, and maximum 11.56, 0, 36 years in service respectively


In [18]:
# Calculating mean values, grouped by rank, rounded to 2
round(df.groupby(['rank']).mean(), 2)


phd  service     salary
rank                                
AssocProf  15.08    11.31   91786.23
AsstProf    5.05     2.21   81362.79
Prof       27.07    21.41  123624.80

Deductions from above output:
Individuals of rank Prof with high phd and service years earn more than their counterparts of other ranks on the average.  

In [19]:
# Calculating mean values, grouped by discipline, rounded to 2
round(df.groupby(['discipline']).mean(), 2)

phd  service     salary
discipline                           
A           21.53    15.72   98331.11
B           18.14    14.48  116331.79

Deductions from above output:
Individuals of the discipline B earn more on the average than their counterparts of discipline A regardless of phd and service years.

In [20]:
# Calculating mean values, grouped by sex, rounded to 2
round(df.groupby(['sex']).mean(), 2)

phd  service     salary
sex                              
Female  16.51    11.56  101002.41
Male    22.90    18.54  115045.15

Deductions from above output:
Male individuals with more phd and service years earn more on the average than their female colleagues.

In [21]:
# Calculating maximum values, grouped by rank
df.groupby(['rank']).max()

discipline  phd  service   sex  salary
rank                                            
AssocProf          B   26       24  Male  119800
AsstProf           B   11        6  Male   97032
Prof               B   56       51  Male  186960

Deductions from above output:
Individuals of the ranks Prof, under discipline B, with higher phd and service years, who are also male earn the highest salary.  

In [22]:
# Calculating maximum values, grouped by discipline
df.groupby(['discipline']).max()

rank  phd  service   sex  salary
discipline                                  
A           Prof   51       51  Male  155865
B           Prof   56       49  Male  186960

Deductions from above output:
Individual of discipline B, who are of the rank Prof, with higher phd and service years, who are also male earn highest salary

In [23]:
# Calculating maximum values, grouped by sex
df.groupby(['sex']).max()

rank discipline  phd  service  salary
sex                                          
Female  Prof          B   39       36  161101
Male    Prof          B   56       51  186960

Deductions from above output:
Male indiduals who are of the rank Prof, under discipline B with higher phd and service years, earn the highest salaries

In [24]:
# Calculating minimum values, grouped by rank
df.groupby(['rank']).min()

discipline  phd  service     sex  salary
rank                                              
AssocProf          A    9        6  Female   62884
AsstProf           A    1        0  Female   63100
Prof               A   12        0  Female   57800

Deductions from above output:
The lowest salary is earned by an individual of the AssocProf rank, under discipline A, with lower phd and service years, who is also female

In [25]:
# Calculating minimum values, grouped by discipline
df.groupby(['discipline']).min()

rank  phd  service     sex  salary
discipline                                         
A           AssocProf    2        0  Female   57800
B           AssocProf    1        0  Female   71065

Deductions from above output:
The lowest salary is earned by an individual under the A discipline, of the AssocProf rank, with lower phd and service years, and also female.

In [26]:
# Calculating minimum values, grouped by sex
df.groupby(['sex']).min()

rank discipline  phd  service  salary
sex                                               
Female  AssocProf          A    2        0   62884
Male    AssocProf          A    1        0   57800

Deductions from above output:
The lowest salary is earned by a male individual of the AssocProf rank, under A discipline, with lower phd and service years.

General deductions from the analysis of the dataset:
1. Individuals under the B discipline tend to earn more than those under the A discipline.
2. Individuals of the Prof rank tend to earn more than those of other ranks.
3. Individuals of the male sex tend to earn more than their female colleagues most of the time, unless they are under the A discipline, also with lower phd and service years.
4. Individuals with considerable more phd and service years "may" earn more than their colleagues depending on their discipline, rank and sex.